flowauth/frontend/src/RoleScopePicker.cy.jsx

```diff
@@ -1,102 +1,140 @@
+  })
+
+  it("cascades correctly", () => {
+    cy.mount(<RoleScopePicker
+      updateScopes={()=>{}}
+      server_id = {1}
+      role_id = {1}
+    />);
+    // Check admin0 starts indeterminant
+    cy.get("[data-cy=nested-admin0")
+      .find("[data-cy=checkbox]").first()
+      .invoke("prop", "indeterminate", true)
+
+    // Expand child scopes and check click on checkbox works
+    cy.get("[data-cy=nested-admin0]").within(() => {
+      cy.get("[data-cy=chevron]")
+        .click()
+      cy.get("[data-cy=nested-dummy_query]")
+        .find("[data-cy=checkbox]").first()
+        .invoke("prop", "checked", true)
+        .click()
+        .invoke("prop", "checked", false)
+    })
+
+    // Check both child scopes of admin0 change on click
+    cy.get("[data-cy=nested-other_query]")
+      .find("[data-cy=checkbox]").first()
+      .click()
 
-		cy.intercept({
-			method: 'GET',
-			url:'/roles/1/scopes'
-		},
-		[{
-			"id": 3,
-			"name": "admin0:dummy_query:inner_dummy_query",
-		},
-		{
-			"id":1,
-			"name":"admin0:dummy_query:dummy_nested_query"
-		},
-		{
-			"id":7,
-			"name":"get_results"
-		}
-	]
-		).as('getRoleScopes');
-				
-		cy.mount(<RoleScopePicker
-				updateScopes={()=>{}}
-				server_id = {1}
-				role_id = {1}
-			/>);
-		cy.wait("@getServerScopes")
-		cy.wait("@getRoleScopes")
-		cy.get('.rs-picker-toggle-value > span').should("include.text", "admin0 (All), get_results")
-	}),
+    //Check both children being true cause admin0 to become true
+    cy.get("[data-cy=nested-admin0")
+      .find("[data-cy=checkbox]").first()
+      .invoke("prop", "checked", true)
 
-	it('respects deeply nested loading differences', () => {
-		cy.intercept({
-				method: 'GET',
-				url: '/admin/servers/1/scopes', 
-			},
-			[
-				{
-					"enabled": true,
-					"id": 1,
-					"name": "admin0:dummy_query:other_query"
-				},
-				{
-					"enabled": true,
-					"id": 3,
-					"name": "admin0:dummy_query:dummy_query"
-				}
-			]
-		).as('getServerScopes');
+    //Click on admin0's checkbox
+    cy.get("[data-cy=nested-admin0")
+      .find("[data-cy=checkbox]").first()
+      .click()
 
-		cy.intercept({
-			method: 'GET',
-			url:'/roles/1/scopes'
-		},
-		[{
-			"id": 3,
-			"name": "admin0:dummy_query:dummy_query",
-		},
-	]
-		).as('getRoleScopes');
-				
-		cy.mount(<RoleScopePicker
-				updateScopes={()=>{}}
-				server_id = {1}
-				role_id = {1}
-			/>);
-		cy.wait("@getServerScopes")
-		cy.wait("@getRoleScopes")
-		cy.get('.rs-picker-toggle-value > span').should("include.text", "admin0:dummy_query:dummy_query")
-	})
-})
+    // Check both children are now false
+    cy.get("[data-cy=nested-admin0]")
+      .find("[data-cy=nested-dummy_query]")
+      .find("[data-cy=checkbox]").first()
+      .invoke("prop", "checked", false)
+    cy.get("[data-cy=nested-admin0]")
+      .find("[data-cy=nested-other_query]")
+      .find("[data-cy=checkbox]").first()
+      .invoke("prop", "checked", false)
+  })
 
+})
```

Comment:
No newline?! 😱
I thought the code formatter takes care of these things?

Reply:
Knew I'd forgotten to do something...